conans/client/conf/__init__.py

```diff
@@ -115,6 +116,12 @@
                     <<: *visual_studio
                 apple-clang:
                     <<: *apple_clang
+        intel-cc:
+            version: ["2021.1", "2021.2", "2021.3"]
+            mode: ["icx", "classic", "dpcpp"]
+            libcxx: [None, libstdc++, libstdc++11, libc++]
+            cppstd: [None, 98, gnu98, 03, gnu03, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20, 23, gnu23]
+            runtime: [None, MD, MT, MTd, MDd]
```

Comment:
Do we want to model the runtime as the ``msvc`` more modern one? (dynamic/static)? Cause it is less confusing and it needs to keep in sync with build_type.